### ISSUE TITLE
Azure Log Analytics scaler

### ISSUE THREAD

**Issue by spoplavskiy** [kedacore/keda#1061]
In standard AKS setup, Log Analytics takes central part in storing OMS agent metrics, logs from pods, during building a monitoring dashboards and generate alerts. Also, OMS agent could be configured to scrap Prometheus metrics, so they also can be landed in Log Analytics workspace. Advantage of using Container Insights (OMS agents) is that "It just works", you don't need to support it, and in many cases even configure it. So, with Log Analytics and Container Insights there is complete monitoring story in place, but the question is, how to use this data to scale deployments? And as of now, the answer is - you can't do this.

To fill this gap, I have started to develop Log Analytics scaler and here I would like to use community experience and Knowles to better understand potential requirements and limitations.

### Use-Case

- Use Log Analytics query to retrieve required metric will be used by Keda and HPA to scale deployments.
- Potentially, use Log Analytics query to retrieve threshold will be configured during scaler creation.

### Requirements and limitations

- Authorisation using AAD Service Principal
- Automatic tracking token expiration and reclaim it, when needed.
- Query should return the following datatypes: real, int, long (use [this](https://i2.wp.com/www.stefanroth.net/wordpress/wp-content/uploads/2018/05/image.png?ssl=1) as a reference for LA datatypes). This will be a part of query result validation.
- Query should return one table with one row.
- Cell # 1 will be used as metric, cell # 2 can be used as threshold (optional). So, in a moment of creating instance of scaler, threshold could be retrieved from ScalerObject metadata, or from LA query result. 

If you have any good proposals\suggestions for LA scaler, feel free to put them in comments.

Thank you!

**Comment by tomkerkhove** [comment]
I like the idea! Are you willing to contribute this?

**Comment by spoplavskiy** [comment]
> I like the idea! Are you willing to contribute this?

Yep, the first version to try is here: https://github.com/kedacore/keda/pull/1085

**Comment by tomkerkhove** [comment]
This is done, thanks man!
